File: 20180717-java-bank/src/main/java/AccountService.java
import java.util.ArrayList;
import java.util.List;

public class AccountService {
    private final Console console;
    private final Calendar calendar;

    private List<String> transactions = new ArrayList<>();

    public AccountService(Console console, Calendar calendar) {
        this.console = console;
        this.calendar = calendar;
    }

    public void deposit(int amount) {
        this.transactions.add("20/10/2017 | " + amount + " | " + amount);
    }

    public void withdraw(int amount) {
    }

    public void printStatement() {
        console.printLine("DATE | AMOUNT | BALANCE");
        for (String transaction : transactions) {
            console.printLine(transaction);
        }
    }
}

File: 20180723-kotlin-bank/src/main/kotlin/AccountService.kt
class AccountService(
        private val console: Console,
        private val calendar: Calendar
) {

    private lateinit var transactions: Transactions

    constructor(console: Console, calendar: Calendar, transactions: Transactions) : this(console, calendar) {
        this.transactions = transactions
    }

    fun deposit(amount: Int) {
        transactions.register(
                Transaction(date = calendar.currentDate(), amount = amount, balance = amount)
        )
    }

    fun withdraw(amount: Int) {
        transactions.register(
                Transaction(
                        date = calendar.currentDate(),
                        amount = -amount,
                        balance = transactions.lastBalance() - amount)
        )
    }

    fun printStatement() {
    }
}